### PR TITLE
Create Modality model from Abbreviation

### LINKED ISSUE #490: Parse modality from abbreviation
# User story
As a user, I want to create a modality model just from abbreviation. 

# Acceptance criteria
- [x] Given an input abbreviation, then modality model is created. 
- [x] Add a unit test that verifies that abbreviations are unique.
- [x] Add a unit test that checks that experiment type string values are unique.

# Sprint Ready Checklist 
- [x] 1. Acceptance criteria defined 
- [x] 2. Team understands acceptance criteria 
- [x] 3. Team has defined solution / steps to satisfy acceptance criteria 
- [x] 4. Acceptance criteria is verifiable / testable 
- [x] 5. External / 3rd Party dependencies identified
- [x] 6. Ticket is prioritized and sized

# Notes
*Add any helpful notes here.*

### PR DESCRIPTION
**Pull request by mekhlakapoor**
closes #490 

creates the modality model from abbreviation key if the key exists. otherwise, it'll create it from the attribute name. 
also has checks to ensure that there are unique abbreviation keys and unique experiment types.

### CODE REVIEW

File: tests/test_data_description.py [hunk 1]
```diff
@@ -164,10 +164,25 @@ def test_parse_name(self):
...
 
     def test_modality_enums(self):
-        """Tests that BaseName enums can be constructed from attr names"""
+        """Tests that BaseName enums can be constructed from abbreviations"""
         self.assertEqual(Modality.ECEPHYS, Modality("ECEPHYS"))
         self.assertEqual(Modality.ECEPHYS, Modality("ecephys"))
 
+    def test_unique_modality_abbreviations(self):
+        """Tests that abbreviations are unique"""
+        abbreviations = []
```
Comment (jtyoung84) [inline tests/test_data_description.py:173]: You probably do this in two lines using list comprehensions:
```
abbreviations = [m.value.abbreviation for m in Modality]
self.assertEqual(len(set(abbreviations)), len(abbreviations))
```

File: tests/test_data_description.py [hunk 2]
```diff
@@ -164,10 +164,25 @@ def test_parse_name(self):
             toks = DerivedDataDescription.parse_name(self.BAD_NAME)
 
     def test_modality_enums(self):
-        """Tests that BaseName enums can be constructed from attr names"""
+        """Tests that BaseName enums can be constructed from abbreviations"""
         self.assertEqual(Modality.ECEPHYS, Modality("ECEPHYS"))
```
Comment (jtyoung84) [inline tests/test_data_description.py:168]: It might be useful to add a few more examples since ecephys isn't too representative of some of the edge cases

File: src/aind_data_schema/data_description.py [hunk 3]
```diff
@@ -103,7 +103,27 @@ class Group(Enum):
...
 
 
-class Modality(Enum, metaclass=BaseNameEnumMeta):
+class ModalityEnumMeta(EnumMeta):
+    """Allows to create complicated enum based on modality abbreviation."""
+
+    def __call__(cls, value, *args, **kw):
+        """Allow enum to be set by a string."""
+        if isinstance(value, str):
+            abbr = dict([(cls[m].value.abbreviation, cls[m]) for m in cls.__members__])
```
Comment (jtyoung84) [inline src/aind_data_schema/data_description.py:112]: I forgot it's possible to loop through enums directly. It might be simpler to construct the dictionary like this:
```
abbr = dict([(m.value.abbreviation, m.name) for m in Modality])
```

Comment (mekhlakapoor) [reply src/aind_data_schema/data_description.py:112]: it doesn't work as a class method

**Review by jtyoung84 — APPROVED** [review]

**Review by dyf — APPROVED** [review]
